scipy/spatial/_geometric_slerp.py

```diff
@@ -184,8 +185,16 @@ def geometric_slerp(start,
                          "both be in at least two-dimensional "
                          "space")
 
+    if t.size == 0:
+        return np.empty((0, start.size))
+
+    if t.min() < 0 or t.max() > 1:
+        raise ValueError("interpolation parameter must be in [0, 1]")
+
+    # Ensure that the output is broadcasted to the correct shape
     if np.array_equal(start, end):
-        return [start] * np.asarray(t).size
+        shape = t.shape + start.shape
+        return np.full(shape, start)
```

Comment:
While I agree that returning the view-list with two array inputs should be fixed (nice find), I do not agree that ignoring the shape of `t` is a bug.

Indeed, it was my intention from the start that the interpolation parameter `t` be treated (effectively) as a float or 1D array of values between which interpolation should happen.

It is also encoded in the docstring return value, which intentionally uses `t.size` in shape of the output to indicate that the dimensionality on `t` is effectively discarded:

![image](https://user-images.githubusercontent.com/7903078/120738656-72d6d400-c4ad-11eb-83e3-dda597705534.png)

Are there really compelling use cases to change the dosctring/intended design because people in the real world are going to use `[[[[0,1]]]]` instead of `[0, 1]`? At least in my view, those obscure/advanced use cases (what is the use case?) could reshape on their own if they *really* need that.

Reply:
> I do not agree that ignoring the shape of `t` is a bug.

The main reason why I'd argue this a bug is that the normal code path, for non-equivalent inputs, _does_ take into account the shape of `t`. In my opinion either option is fine (ignoring or respecting the shape of `t`), but consistency is important here. Namely, the shape of the output array-like is currently not just determined by the input shapes, but also the input values, which is a rather questionable pattern.

An alternative would be to flatten `t` in all cases (or raise an exception?), but the backwards compatibility concerns of this approach would be greater.